lib/config.js

```diff
@@ -1,14 +1,43 @@
+'use strict';
+
 var config = {
-    brand: 'Discord',
-    token: process.env.OAUTH_TOKEN,
-    trackingID: process.env.TRACKING_ID,
-    commentWait: process.env.COMMENT_WAIT || 250,
-    commentAttempts: process.env.COMMENT_ATTEMPTS || 10,
-
-    // Used when no environment variable (process.env.PORT) is set
-    port: 3000,
-    host: 'localhost',
-    protocol: 'http:'
+    def: {
+        brand: 'Discord',
+        token: process.env.OAUTH_TOKEN,
+        commentAttempts: process.env.COMMENT_ATTEMPTS || 10,
+    },
+    development: {
+        port: 3000,
+        redisURL: 'redis://localhost',
+        databaseURL: 'postgres://localhost/discord_test'
+    },
+    test: {
+        protocol: 'http:',
+        host: 'localhost',
+        port: 3000,
+        trackingID: '654321',
+        commentWait: 0,
+        redisURL: 'redis://localhost',
+        databaseURL: 'postgres://localhost/discord_test',
+        database: 'discord_test'
+    },
+    production: {
+        port: process.env.PORT,
+        trackingID: process.env.TRACKING_ID,
+        commentWait: process.env.COMMENT_WAIT || 250,
+        redisURL: process.env.REDIS_URL,
+        databaseURL: process.env.DATABASE_URL
+    }
```

Comment:
Awesome. Thanks for putting so much thought into that.

That's an interesting approach. It would be much more concise. In terms of 12factor, would that be much different than using environment groups? Both are be forms of in-source configuration.